src/main.c

```diff
@@ -97,6 +100,60 @@ unsigned long AuxFragBufOffset = 0;
 DOT11_EXTSTA_RECV_CONTEXT PacketMetadata;
 BOOLEAN AddMetadata = FALSE;
 
+typedef struct _NDIS_NET_BUFFER_LIST_8021Q_INFO {
+    union {
+        struct {
+            UINT32 UserPriority : 3;             // 802.1p priority
+            UINT32 CanonicalFormatId : 1;        // always 0
+            UINT32 VlanId : 12;                  // VLAN Identification
+            UINT32 Reserved : 16;                // set to 0 for ethernet
+        } TagHeader;
+
+        struct {
+            UINT32 UserPriority : 3;             // 802.1p priority
+            UINT32 CanonicalFormatId : 1;        // always 0
+            UINT32 VlanId : 12;                  // VLAN Identification
+            UINT32 WMMInfo : 4;
+            UINT32 Reserved : 12;                // set to 0 for Wireless LAN
+        } WLanTagHeader;
+
+        PVOID Value;
+    };
+} NDIS_NET_BUFFER_LIST_8021Q_INFO, *PNDIS_NET_BUFFER_LIST_8021Q_INFO;
+
+#define MaxNetBufferListInfo 28
```

Comment:
Where does the number 28 come from?

Reply:
It comes from the enum _NDIS_NET_BUFFER_LIST_INFO:
dt ndis!_NDIS_NET_BUFFER_LIST_INFO
   TcpIpChecksumNetBufferListInfo = 0n0
....
   MaxNetBufferListInfo = 0n28

Thanks for the remark, actually that MaxNetBufferListInfo might change (NDIS version, etC...):
https://docs.microsoft.com/en-us/windows-hardware/drivers/ddi/nblinfo/ne-nblinfo-ndis_net_buffer_list_info

As we cannot predict, what is the NDIS version on the system where the trace comes from, a bigger value is safe.
I would take 64 to be comfortable with NDIS update. 

An alternative is to allocate dynamically the right size of OobData buffer by using OOBDataSize, but that would add a malloc() call in hot path... I don't think this is what we want...

Reply:
It's pretty cheap to make this significantly bigger, so I've bumped it up to 200 and made the printf message very clear about what's wrong when we encounter an OOB data that's too big.